### PR TITLE
Add minio objectbucket

### PR DESCRIPTION
**Pull request by Kidswiss**
This will add an additional `ObjectBucket` implementation backed by Minio.

Also changes to make it integrate more seamlessly with the minio provisioning.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -335,6 +348,13 @@ parameters:
               secretNamespace: ${appcat:services:generic:objectstorage:secretNamespace}
               providerSecretNamespace: ${appcat:providers:cloudscale:connectionSecretNamespace}
               bucketDeletionPolicy: ${appcat:services:generic:objectstorage:bucketDeletionPolicy}
+            minio:
+              enabled: false
+              secretNamespace: ${appcat:services:generic:objectstorage:secretNamespace}
+              providerSecretNamespace: ${appcat:providers:cloudscale:connectionSecretNamespace}
+              bucketDeletionPolicy: ${appcat:services:generic:objectstorage:bucketDeletionPolicy}
+              grpcEndpoint: ${appcat:grpcEndpoint}
```
Comment (zugao) [inline class/defaults.yml:356]: Do we really need the GRPC endpoint to be exposed?

File: component/provider.jsonnet [hunk 2]
```diff
@@ -304,4 +304,56 @@ local controllerConfigRef(config) =
       role,
       rolebinding,
     ],
+  [if params.providers.minio.enabled then '10_provider_minio']:
```
Comment (zugao) [inline component/provider.jsonnet:307]: If I'm not wrong we decided to include everything in composition functions.

**Review by zugao — COMMENTED** [review]
Some things...

File: class/defaults.yml (hunk 1 above)
Comment (Kidswiss) [reply class/defaults.yml:356]: I usually set it in `vshn.yaml` when debugging, then I don't have to set it every time after I re-generate.

File: component/provider.jsonnet (hunk 2 above)
Comment (Kidswiss) [reply component/provider.jsonnet:307]: The provider itself is not deployed by a composition. It's deployed like all other providers via the Crossplane package.

Comment (zugao) [reply component/provider.jsonnet:307]: Ah I missed that. Good then.

**Review by zugao — COMMENTED** [review]
LGTM

**Comment by Kidswiss** [comment]
@zugao can you please press the approve button then? :D

**Review by zugao — APPROVED** [review]

**Comment by zugao** [comment]
Done, sry :D

File: class/defaults.yml [hunk 3]
```diff
@@ -39,7 +39,7 @@ parameters:
       appcat:
         registry: ghcr.io
         repository: vshn/appcat
-        tag: v4.25.0
+        tag: add/minio-providerconfig
```
Comment (Kidswiss) [inline class/defaults.yml:42]: Will be adjusted once appcat is released and tagged.

**Review by wejdross — APPROVED** [review]
overall looks good, please remember about class/defaults.yaml

**Pull request merged**